src/commands/test/lib/included-files-parser.ts

```diff
@@ -1,12 +1,63 @@
-import { IFileDescriptionJson } from "./test-manifest-reader";
+import { IFileDescriptionJson, ITestCloudManifestJson } from "./test-manifest-reader";
 import * as path from "path";
+import * as pfs from "../../../util/misc/promisfied-fs";
+import { validXmlFile } from "./xml-util";
+import { out } from "../../../util/interaction";
+import _ = require("lodash");
 
 const invalidCharactersRegexp = /['"!#$%&+^<=>`|]/;
 
+export async function processIncludedFiles(manifest: ITestCloudManifestJson, include: string[], rootDir: string) {
+  if (!include) {
+    return;
+  }
+
+  const filteredFiles = this.filterIncludedFiles(manifest.files, include);
+  const includedFiles = this.parseIncludedFiles(filteredFiles, rootDir);
+  await this.copyIncludedFiles(manifest, includedFiles, rootDir);
+}
+
+export async function copyIncludedFiles(manifest: ITestCloudManifestJson, includedFiles: IFileDescriptionJson[], rootDir: string) {
+  for (const includedFile of includedFiles) {
+    const copyTarget = path.join(path.dirname(rootDir), includedFile.targetPath);
+    await pfs.cp(includedFile.sourcePath, copyTarget);
+    manifest.files.push(includedFile.targetPath);
+  }
+}
+
 export function parseIncludedFiles(includedFiles: string[], rootDir: string): IFileDescriptionJson[] {
   return includedFiles.map((f) => parseIncludedFile(f, rootDir));
 }
 
+export function filterIncludedFiles(manifestFiles: string[], include: string[]): string[] {
+  if (!include) {
+    return [];
+  }
+
+  const allIncludedFiles = manifestFiles.concat(include);
+  const output = [];
+  for (const file of include) {
+    if (validFile(file, allIncludedFiles)) {
+      output.push(file);
+    }
+  }
```

Comment:
lines 38-45 Could be replaced with `return allIncludedFiles.filter((f) => validFile(file, allIncludedFiles));`

Reply:
Done - need to be:
> return include.filter((f) => validFile(f, allIncludedFiles));

We're only filtering on `--include` on this PR - I made the assumption that we only choose files automatically if those files are OK to upload (that certainly seems to be the case when testing the CLI with commands).